src/newfight.cpp

```diff
@@ -1360,3 +1411,43 @@ void engage_close_combat_if_appropriate(struct combat_data *att, struct combat_d
     }
   }
 }
+
+bool handle_flame_aura(struct combat_data *att, struct combat_data *def) {
+  // Now that we've counterstrike-swapped if applicable, check and try to apply flame aura.
+  if (AFF_FLAGGED(def->ch, AFF_FLAME_AURA) || MOB_FLAGGED(def->ch, MOB_FLAMEAURA)) {
+    int force;
+
+    // First, we require that the attacker really is fighting in melee and unarmed. Flame aura only burns body parts.
+    if (att->ranged_combat_mode || !att->weapon) {
+      // Flameaura-flagged NPCs just use their own level as the force.
+      if (MOB_FLAGGED(def->ch, MOB_FLAMEAURA)) {
+        force = GET_LEVEL(def->ch);
+      }
+      else {
+        // Iterate through their spells, find the flame aura that's applied to them, and extract its force.
+        for (struct sustain_data *sust = GET_SUSTAINED(def->ch); sust; sust = sust->next) {
+          if (!sust->caster && sust->spell == SPELL_FLAMEAURA) {
+            force = sust->force;
+            break;
+          }
+        }
+      }
+
+      if (GET_EQ(att->ch, WEAR_HANDS) && GET_EQ(att->ch, WEAR_FEET)) {
```

Comment:
You'd mentioned wanting to use adept powers / spells / etc instead of checking for hands / feet wearslots, this would be the spot to do it in.